torch/nn/modules/utils.py

```diff
@@ -32,3 +32,34 @@ def _list_with_default(out_size: List[int], defaults: List[int]) -> List[int]:
     if len(defaults) <= len(out_size):
         raise ValueError('Input dimension should be at least {}'.format(len(out_size) + 1))
     return [v if v is not None else d for v, d in zip(out_size, defaults[-len(out_size):])]
+
+
+def consume_prefix_in_state_dict_if_present(state_dict: Dict[str, Any], prefix: str):
+    r"""Strip the prefix in state_dict metadata, if any.
+
+    Args:
+        state_dict (OrderedDict): a state-dict to be loaded to the model.
+        prefix (str): prefix.
+    """
+    keys = sorted(state_dict.keys())
+    for key in keys:
+        if key.startswith(prefix):
+            newkey = key[len(prefix) :]
+            state_dict[newkey] = state_dict.pop(key)
+
+    # also strip the prefix in metadata if any.
+    try:
+        metadata = state_dict._metadata  # pyre-ignore
+    except AttributeError:
+        pass
+    else:
+        for key in list(metadata.keys()):
+            # for the metadata dict, the key can be:
+            # '': for the DDP module, which we want to remove.
+            # 'module': for the actual model.
+            # 'module.xx.xx': for the rest.
+
+            if len(key) == 0:
+                continue
+            newkey = key[len(prefix) :]
```

Comment:
Do we not need to check if it startswith prefix here?

Reply:
The comments here assumes that there are only 3 cases: as long as there is a prefix, the prefix must start with "module", then no need to check. This is my understanding.

I copied this code snippet from https://github.com/facebookresearch/fvcore/blob/master/fvcore/common/checkpoint.py#L503

I guess here the original author probably wants to save some overheads of unnecessary string comparison.

Reply:
So every key in `metadata.keys()` is either `""` or starts with `module`? Are we sure that's always true for a DDP model, can we verify experimentally?